packages/headless/src/features/product-listing/product-listing-actions.ts

```diff
@@ -43,6 +44,24 @@ export const setProductListingUrl = createAction(
     })
 );
 
+export interface SetAdditionalFieldsPayload {
+  /**
+   * The additional fields to fetch with the product listing.
+   */
+  additionalFields: string[];
+}
+
+export const setAdditionalFields = createAction(
+  'productlisting/setAdditionalFields',
+  (payload: SetAdditionalFieldsPayload) =>
+    validatePayload(payload, {
+      additionalFields: new ArrayValue({
+        required: true,
```

Comment:
Does making this required prevent us from clearing the `additionalFields` by passing an empty array?

Reply:
No, it just makes the parameter required, so you can't pass: `additionalFields: null`